include/c_dataset.h

```diff
@@ -247,6 +247,25 @@ SRError get_tensor_names(
 SRError get_tensor_type(
     void* dataset, const char* name, size_t name_len, SRTensorType* ttype);
 
+/*!
+*   \brief Retrieve the dimensions of a Tensor in the DataSet
+*   \param dataset The dataset to use for this operation
+*   \param name The name of the tensor (null-terminated string)
+*   \param name_len The length in bytes of the tensor name
+*   \param dims A buffer in which to receive the dimensions for
+*               the specified tensor.
+*   \param ndims The size of the buffer supplied in \p dims when this function
```

Comment:
The behavior of this function seems different than other functions like ``get_tensor_names``.  This function assumes you know the dimensions to be able to return the dims?  Is that because we don't want to handle the memory or is there another reason having to do with Fortran?  Would it make sense just to return ``dims`` for the user so they don't have to worry about it and get ``SRBadAllocError``?

Reply:
Exactly so -- it makes sure that we can avoid leaking memory by having the caller supply the buffer into which we copy the dims. If we just returned the dims, we'd have to allocate memory and then give the caller an interface through which to say they don't need this block any more -- which gets really messy really fast. With this approach, if the caller doesn't provide a big enough buffer, we can tell them how big the buffer needs to be, so worst case, they'll only have to call the method twice.